### PR TITLE
secrets/hashivault: Add support for VAULT_ADDR and VAULT_TOKEN

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #2990

`VAULT_ADDR` and `VAULT_TOKEN` are the recommended way of configuring
access to vault as per the Vault docs https://www.vaultproject.io/docs/commands#environment-variables

This commit adds support for those environment variables. We check
that the *existing* VAULT_SERVER_URL and VAULT_SERVER_TOKEN env vars
are not set first. If they are, then we use those. If they are not
then we fall back to the new environment variables

Please use a title starting with the name of the affected package, or \"all\",
followed by a colon, followed by a short summary of the issue. Example:
`blob/gcsblob: fix typo in documentation`.

Please reference any Issue related to this Pull Request. Example: `Fixes #1`.

See
[here](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)
for tips on good Pull Request description.

### CODE REVIEW

**Comment by stack72** [comment]
@googlebot I signed it!

File: secrets/hashivault/vault.go [hunk 1]
```diff
@@ -74,8 +74,41 @@ func init() {
 	secrets.DefaultURLMux().RegisterKeeper(Scheme, new(defaultDialer))
 }
 
+// getVaultUrl ensures that we check both VAULT_SERVER_URL and VAULT_ADDR environment
+// variables for the API address for vault. VAULT_SERVER_URL takes precedent over VAULT_ADDR
```
Comment (vangent) [inline secrets/hashivault/vault.go:78]: nit: precedence, and end with a ".".

File: secrets/hashivault/vault.go [hunk 2]
```diff
@@ -74,8 +74,41 @@ func init() {
...
+func getVaultUrl() (string, error) {
```
Comment (vangent) [inline secrets/hashivault/vault.go:79]: nit: getVaultURL

File: secrets/hashivault/vault.go [hunk 3]
```diff
@@ -74,8 +74,41 @@ func init() {
...
+	if serverURL != "" {
+		return serverURL, nil
+	}
+
+	vaultAddr := os.Getenv("VAULT_ADDR")
+	if vaultAddr != "" {
+		return vaultAddr, nil
+	}
+
+	return "", errors.New("neither VAULT_SERVER_URL or VAULT_ADDR environment variables are set")
```
Comment (vangent) [inline secrets/hashivault/vault.go:90]: nit: or -> nor

File: secrets/hashivault/vault.go [hunk 4]
```diff
@@ -74,8 +74,41 @@ func init() {
...
+}
+
+// getVaultToken ensures that we check both VAULT_SERVER_TOKEN and VAULT_TOKEN environment
+// variables for the API token for vault. VAULT_SERVER_TOKEN takes precedent over VAULT_TOKEN
```
Comment (vangent) [inline secrets/hashivault/vault.go:94]: Same comments as above.

File: secrets/hashivault/vault.go [hunk 5]
```diff
@@ -74,8 +74,41 @@ func init() {
 	secrets.DefaultURLMux().RegisterKeeper(Scheme, new(defaultDialer))
 }
 
+// getVaultUrl ensures that we check both VAULT_SERVER_URL and VAULT_ADDR environment
```
Comment (vangent) [inline secrets/hashivault/vault.go:77]: Please update the package docstring near line 23.

**Comment by stack72** [comment]
Thanks for the review here @vangent 

All changes made!

**Comment by vangent** [comment]
FYI, the Travis build has gotten flaky recently (Docker failures trying to bring up local services in containers), not sure why. And, I can't restart/retry the Travis test runs anymore due to some hiccup with the Google org and Travis. I hope this is resolved soon. If you'd like, you can try to push an empty commit and force a retry that way (but it might take a few tries, so if you'd rather just wait that's fine too).

**Comment by stack72** [comment]
Hey @vangent 

done ;)

**Review by vangent — APPROVED** [review]

**Pull request merged**